Automattic/wp-calypso
# FSE: Add a navigation placeholder.

#### Changes proposed in this Pull Request

Add a navigation placeholder to represent active menus within an FSE header template part.

**Main Menu**
<img width="831" alt="Screen Shot 2019-06-06 at 2 36 16 PM" src="https://user-images.githubusercontent.com/349751/59068235-7efd0c00-8868-11e9-9e3d-f5c9e1812345.png">

**Social Menu**
<img width="841" alt="Screen Shot 2019-06-06 at 2 35 26 PM" src="https://user-images.githubusercontent.com/349751/59068243-845a5680-8868-11e9-963a-f46f278f1d3f.png">

**Footer Menu**
<img width="861" alt="Screen Shot 2019-06-06 at 2 27 39 PM" src="https://user-images.githubusercontent.com/349751/59068251-89b7a100-8868-11e9-8a21-0549add409b4.png">


#### Testing instructions

* Load your FSE testing environment with this branch active.
* Add a "Navigation Placeholder" block anywhere you like (it's not yet scoped to anything in particular).
* Verify that the Edit preview represents the theme's one-level menu.
* Verify the front-end displays the one-level menu as expected.
* Enter code view and add another menu type, does this reflect the created menu. Is this read correctly on a second editor load?
```


```

Fixes #33109 .
